### PR TITLE
add document for offline installation

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
Signed-off-by: xiang <xiang13225080@163.com>

### CODE REVIEW

**Comment by g1eny0ung** [comment]
Does it need to load CRD before installation?

**Comment by WangXiangUSTC** [comment]
> Does it need to load CRD before installation?

https://github.com/chaos-mesh/chaos-mesh/pull/1200
after this pr merged, crd in already contained in the helm chart @g1eny0ung

**Comment by g1eny0ung** [comment]
> > Does it need to load CRD before installation?
> 
> [chaos-mesh/chaos-mesh#1200](https://github.com/chaos-mesh/chaos-mesh/pull/1200)
> after this pr merged, crd in already contained in the helm chart @g1eny0ung

Great! 🎉

**Review by g1eny0ung — APPROVED** [review]
LGTM

File: docs/user_guides/offline_installation.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
+---
+id: offline_installation
+title: Offline Installation
+---
+
+This document describes how to install Chaos Mesh under an offline environment, which means can not connect to the internet.
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply docs/user_guides/offline_installation.md]: The production environment of some state-owned enterprises has strict network requirements and maybe cannot connect to external networks

**Comment by WangXiangUSTC** [comment]
@dcalvin @cwen0 PTAL again

File: docs/user_guides/offline_installation.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+### Copy image and repo archive files
+
+Copy the file `./image-chaos-mesh`, `./image-chaos-daemon`, `./image-chaos-dashboard` and `{CHAOS_MESH_VERSION}.zip` into the offline environment.
+
+## Install Chaos Mesh
+
+Now you already have the image and repo archive files in the offline environment, you can use them to install Chaos Mesh.
+
+Specify the version you are going to install in the offline environment:
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply docs/user_guides/offline_installation.md:68]: The first one is executed in the computer which can connect to the network, this one is executed in the offline environment.

[the remark this answers is not in the document]
Comment (dcalvin) [reply docs/user_guides/offline_installation.md:65]: Users need to connect to the network to prepare the installation files, with a specified version - this means the version is already specified for the installation, right?

[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply docs/user_guides/offline_installation.md:66]: Yes, but need to execute twice

File: docs/user_guides/offline_installation.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+You need to prepare the installation file with a computer connected to the internet. 
+
+You can specify which version you want to install:
+
+```bash
+export CHAOS_MESH_VERSION="v1.1.0"
+```
+
+It is recommended that you use a stable release, or you can set the version to `latest` if you want to experience the latest features that are under development.
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply docs/user_guides/offline_installation.md]: done

File: docs/user_guides/offline_installation.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+### Download Chaos Mesh repo
+
+Download Chaos Mesh repo with release version:
+
+```bash
+wget "https://github.com/chaos-mesh/chaos-mesh/archive/${CHAOS_MESH_VERSION}.zip"
+```
+
+Or you can download the latest unstable version:
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply docs/user_guides/offline_installation.md:53]: For the latest images, the image name is `...:latest`, but for the repo, it is `.../master.zip`, so I description it again.

File: docs/user_guides/offline_installation.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (dcalvin) [reply docs/user_guides/offline_installation.md:66]: Maybe something to consider for improvement - this sounds unnecessary

File: docs/user_guides/offline_installation.md [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+First, you need to pull images of Chaos Mesh:
+
+```bash
+docker pull pingcap/chaos-mesh:${CHAOS_MESH_VERSION}
+docker pull pingcap/chaos-daemon:${CHAOS_MESH_VERSION}
+docker pull pingcap/chaos-dashboard:${CHAOS_MESH_VERSION}
+```
+
+Then save these images to tar archives:
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply docs/user_guides/offline_installation.md:39]: done

File: docs/user_guides/offline_installation.md [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+
+Load the image from archive files:
+
+```bash #load images of Chaos Mesh
+docker load -i ./image-chaos-mesh
+docker load -i ./image-chaos-daemon
+docker load -i ./image-chaos-dashboard
+```
+
+You can choose to push these images to Docker Registry or Docker Hub.
```
Comment (dcalvin) [inline docs/user_guides/offline_installation.md:82]:
```suggestion
3. Push the Chaos Mesh images. You can choose to push them to Docker Registry or Docker Hub.
```

File: docs/user_guides/offline_installation.md [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+
+   Then push these images to Docker Registry:
+
+   ```bash #push tags of Chaos Mesh to Docker hub
+   docker push $CHAOS_MESH_IMAGE
+   docker push $CHAOS_DAEMON_IMAGE
+   docker push $CHAOS_DASHBOARD_IMAGE
+   ```
+
+2. Installation
```
Comment (dcalvin) [inline docs/user_guides/offline_installation.md:142]:
```suggestion
2. Install Chaos Mesh offline with the following steps:
```

File: docs/user_guides/offline_installation.md [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+
+   a. Unzip the repo archive files:
+
+   ```bash #unzip archive file of repo
+   unzip {CHAOS_MESH_VERSION}.zip chaos-mesh
+   ```
+
+   And then enter the directory.
```
Comment (dcalvin) [inline docs/user_guides/offline_installation.md:150]: Please keep all the step info in one place.

Comment (dcalvin) [inline docs/user_guides/offline_installation.md:150]:
```suggestion
```

File: docs/user_guides/offline_installation.md [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+
+   ```bash #push tags of Chaos Mesh to Docker hub
+   docker push $CHAOS_MESH_IMAGE
+   docker push $CHAOS_DAEMON_IMAGE
+   docker push $CHAOS_DASHBOARD_IMAGE
+   ```
+
+2. Installation
+
+   a. Unzip the repo archive files:
```
Comment (dcalvin) [inline docs/user_guides/offline_installation.md:144]:
```suggestion
   a. Unzip the repo archive files to a path:
```

File: docs/user_guides/offline_installation.md [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+
+   ```bash #unzip archive file of repo
+   unzip {CHAOS_MESH_VERSION}.zip chaos-mesh
+   ```
+
+   And then enter the directory.
+
+   b. Create a namespace for install Chaos Mesh:
```
Comment (dcalvin) [inline docs/user_guides/offline_installation.md:152]:
```suggestion
   b. Create a namespace for installing Chaos Mesh:
```

File: docs/user_guides/offline_installation.md [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+
+   ```bash #create a namespace for install Chaos Mesh
```
Comment (dcalvin) [inline docs/user_guides/offline_installation.md:154]:
```suggestion
   ```bash  
```

File: docs/user_guides/offline_installation.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply docs/user_guides/offline_installation.md:52]: We need them all, these two steps download different files.

File: docs/user_guides/offline_installation.md [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+   docker load -i ./image-chaos-daemon
+   docker load -i ./image-chaos-dashboard
+   ```
+
+
+3. Push the Chaos Mesh images. You can choose to push them to Docker Registry or Docker Hub.
+
+- Push images to Docker Registry
+
+   Set the Docker Registry variable, for example:
```
Comment (dcalvin) [inline docs/user_guides/offline_installation.md:84]:
```suggestion
   a. Set the Docker Registry variable, for example:
```

Comment (dcalvin) [reply docs/user_guides/offline_installation.md:84]: Add sequence label for these steps.

File: docs/user_guides/offline_installation.md [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+```
+
+> **Note:**
+>
+> It is recommended that you use a stable release, or you can set the version to `latest` if you want to experience the latest features that are under development.
+
+2. Archive the docker images of Chaos Mesh:
+
+
+```bash  #pull images of Chaos Mesh
```
Comment (dcalvin) [inline docs/user_guides/offline_installation.md:33]: Indent needed here and below

Comment (WangXiangUSTC) [reply docs/user_guides/offline_installation.md]: done

File: docs/user_guides/offline_installation.md (hunk 12 above)
Comment (WangXiangUSTC) [reply docs/user_guides/offline_installation.md]: done

File: docs/user_guides/offline_installation.md [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+   ```bash
+   docker load -i ./image-chaos-mesh
+   docker load -i ./image-chaos-daemon
+   docker load -i ./image-chaos-dashboard
+   ```
+
+
+3. Push the Chaos Mesh images. You can choose to push them to Docker Registry or Docker Hub.
+
+- Push images to Docker Registry
```
Comment (dcalvin) [inline docs/user_guides/offline_installation.md]:
```suggestion
    - Push images to Docker Registry
```

Comment (dcalvin) [reply docs/user_guides/offline_installation.md]: More indent here.

**Review by dcalvin — APPROVED** [review]
LGTM

**Pull request merged**